internal/rotate.go

```diff
@@ -35,49 +24,27 @@ type CertRotationState struct {
 	Finalized bool
 }
 
-type CertRotationHandler struct {
-	State     CertRotationState
-	stateFile string
-	app       *App
-	client    *http.Client
-	crypto    *EciesCrypto
-	eventSync EventSync
-	steps     []CertRotationStep
-	cienv     bool
-}
+type certRotationContext = stateContext[*CertRotationState]
+type certRotationStep = stateStep[*CertRotationState]
 
-type CertRotationStep interface {
-	Name() string
-	Execute(handler *CertRotationHandler) error
+// Not type CertRotationHandler stateHandler[*CertRotationState].
+// We want methods from a parent to be inherited, thus use struct composition.
+type CertRotationHandler struct {
+	stateHandler[*CertRotationState]
 }
```

Comment:
Using `type CertRotationHandler = stateHandler[*CertRotationState]` would not allow to add new methods, as this would not be a new class.
Using `type CertRotationHandler stateHandler[*CertRotationState]` would not inherit methods from the base class, as this is a type override.

Only a struct composition like done above both inherits base methods and allows to add more methods.